backend/src/test/java/gov/cdc/usds/simplereport/api/ApiUserManagementTest.java

```diff
@@ -83,69 +89,191 @@ void whoami_nobody_okResponses() {
+        when(_oktaRepo.getAllUsernamesForOrganization(argThat(defaultOrgMatcher), eq(OrganizationRole.ADMIN)))
+                .thenReturn(List.of());
+        when(_oktaRepo.getAllUsernamesForOrganization(argThat(defaultOrgMatcher), eq(OrganizationRole.ENTRY_ONLY)))
+                .thenReturn(List.of());
+      
+        ObjectNode resp = runQuery("users-query");
+        List<ObjectNode> usersRetrieved = toList((ArrayNode) resp.get("users"));
+        
+        assertEquals(usersRetrieved.size(), usersAdded.size());
+
+        Collections.sort(usersAdded, new SortByEmail());
+        Collections.sort(usersRetrieved, new SortByEmail());
+
+        for (int i = 0; i < usersRetrieved.size(); i++) {
+            ObjectNode userRetrieved = (ObjectNode) usersRetrieved.get(i);
+            ObjectNode userAdded = usersAdded.get(i);
+            assertEquals(userRetrieved.get("firstName").asText(),
+                         userAdded.get("firstName").asText());
+            assertEquals(userRetrieved.get("email").asText(),
+                         userAdded.get("email").asText());
+            assertEquals(userRetrieved.get("organization").get("externalId").asText(),
+                         userAdded.get("organizationExternalId").asText());
+            assertEquals(OrganizationRole.USER.getGrantedPermissions(), 
+                         extractPermissionsFromUser(userRetrieved));
+        }
+    }
+
+    @Test
+    void getUsers_orgUser_failure() {
+        runQuery("users-query", ACCESS_ERROR);
+    }
+
+    private class OrganizationMatcher implements ArgumentMatcher<Organization> {
+        private Organization left;
+
+        public OrganizationMatcher(Organization left) {
+            this.left = left;
+        }
+
+        public boolean matches(Organization right) {
+            return left == null && right == null ||
+                   left != null && right != null &&
+                   left.getExternalId().equals(right.getExternalId()) &&
+                   left.getOrganizationName().equals(right.getOrganizationName());
+        }
+    }
+
+    private class SortByEmail implements Comparator<ObjectNode> 
+    { 
+        // Used for sorting by email
+        public int compare(ObjectNode userA, ObjectNode userB) 
+        { 
+            return userA.get("email").asText().compareTo(userB.get("email").asText()); 
+        } 
+    } 
+
+    private List<ObjectNode> toList(ArrayNode arr) {
+        List<ObjectNode> list = new ArrayList<>();
+        for (int i = 0; i < arr.size(); i++) {
+            list.add((ObjectNode) arr.get(i));
+        }
+        return list;
+    }
+
+    private OrganizationRoleClaims getOrgRoles(OrganizationRole role) {
+        Set<OrganizationRole> roles = new HashSet<>();
+        roles.add(OrganizationRole.USER);
+        roles.add(role);
+        return new OrganizationRoleClaims(_initService.getDefaultOrganization().getExternalId(),
+                                                   roles);
```

Comment:
```suggestion
        return new OrganizationRoleClaims(_initService.getDefaultOrganization().getExternalId(),
                                                   Set.of(OrganizationRole.USER, role));
```

Or not, honestly, this is fine. 🙂

Reply:
needs to be able to de-dupe duplicates though! Set can't, but HashSet can.

Reply:
Really? Set.of chokes on duplicates? I would not have guessed that! (Obviously, I suppose!)